### DISCUSSION TITLE
choppy TripsLayer rendering

### DISCUSSION THREAD

**Discussion by charlieforward9** [visgl/deck.gl#9260] [Q&A]
https://github.com/user-attachments/assets/3d5c59e7-48e3-47e1-b24d-e8704ea94a2a

I am experiencing strange behavior with my trip layers. While I try to optimize the code, does any of these metrics seem like a clear culprit?

```
{
    "fps": 59.995932480608516,
    "setPropsTime": 1.9000000059604645,
    "updateAttributesTime": 29.399999976158142,
    "framesRedrawn": 60,
    "pickTime": 0,
    "pickCount": 0,
    "gpuTime": 0,
    "gpuTimePerFrame": 0,
    "cpuTime": 113.09999993443489,
    "cpuTimePerFrame": 1.8849999989072481,
    "bufferMemory": 0,
    "textureMemory": 0,
    "renderbufferMemory": 0,
    "gpuMemory": 0
}
```

**Comment by Pessimistress** [comment]
It looks like you are not updating `currentTime` smoothly?

**Reply by charlieforward9** [reply]
Im trying everything I can

https://github.com/user-attachments/assets/8bf9a3c5-2c04-4f9c-abc4-7a37e1058fea

Smoothly managing the currentTime variable in the state

Layer rendering happens with...
```typescript
 const { currentTime, tripLayer ... } = useNH();

 const [layers, setLayers] = useState<LayersList>([
      new Tile3DLayer({
        ...tile3dProps,
      }),
  ]);
  
  useEffect(() => {
    //handle viewport position computation and dispatching

    setLayers([
      new Tile3DLayer({
        ...tile3dProps,
      }),
      new TripsLayer({
        ...layerProps,
        data: tripLayer,
        currentTime,
      }),
    ]);
    console.log("currentTime", t);
  }, [currentTime]);


 return (
     <DeckGL
       style={{ position: "relative" }}
       controller={true}
       layers={layers}
        initialViewState={MCP2MVS(cameraProps)}
       {...((animating || loading) && MCP2MVS(cameraProps))}
     />
 );
``` 

I am hiding some of the other logic to simplify my code. When logging the layer objects, I see that it is storing currentTime properly, its when I pass it into DeckGL, it does not keep up. 

The `tripLayer` has a timestamp value every second

I will try putting it into a online sample in the next few days.
